langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/Index.java

```diff
@@ -0,0 +1,46 @@
+package dev.langchain4j.store.embedding.oracle;
+
+/**
+ * Abstract class for that represents an Index. It is extended by {@link IVFIndex} and
+ * {@link JSONIndex}.
+ */
+public abstract class Index {
+  private CreateOption createOption;
+  private String createIndexStatement;
+  private String dropIndexStatement;
+
+  Index(CreateOption createOption, String createIndexStatement, String dropIndexStatement) {
+    this.createIndexStatement = createIndexStatement;
+    this.dropIndexStatement = dropIndexStatement;
+    this.createOption = createOption;
+  }
+
+  /**
+   * @return The CREATE INDEX statement.
+   */
+  public String getCreateStatement() {
```

Comment:
It seems like these public methods could be package-private instead.
We could even go further with the encapsulation, and replace these methods with a `create(DataSource, EmbeddingTable)` method. That could be nice as subclasses have more flexibility in how they create and drop objects. For instance, they might execute multiple statements. Or maybe we don't need it. Hard to guess what the future holds...